EmeraldWarden_main.lua

```diff
@@ -0,0 +1,342 @@
+	= _G.print, _G.ipairs, _G.pairs, _G.string, _G.table, _G.next, _G.type, _G.table.insert, _G.table.remove, _G.table.sort, _G.string.format, _G.tostring, _G.tonumber, _G.string.find, _G.string.sub
+local ceil, floor, pi, tan, atan, atan2, abs, cos, sin, acos, max, random
+	= _G.math.ceil, _G.math.floor, _G.math.pi, _G.math.tan, _G.math.atan, _G.math.atan2, _G.math.abs, _G.math.cos, _G.math.sin, _G.math.acos, _G.math.max, _G.math.random
+
+local BotEcho, VerboseLog, BotLog = core.BotEcho, core.VerboseLog, core.BotLog
+local Clamp = core.Clamp
+
+BotEcho('loading warden_main...')
+
+--------------------------------
+-- Lanes
+--------------------------------
+core.tLanePreferences = {Jungle = 0, Mid = 5, ShortSolo = 0, LongSolo = 0, ShortSupport = 0, LongSupport = 0, ShortCarry = 5, LongCarry = 3}
+-- Jungle 0 -> Cannot jungle
+-- Mid 5 -> Easier to farm vs one hero
+-- ShortSolo 0 -> Cannot solo because it has no good escape skill
+-- LongSolo 0 -> Cannot solo because it has no good escape skill
+-- ShortSupport 0 -> Cannot provide good support
+-- LongSupport 0 -> Cannot provide good support
+-- ShortCarry 5 -> Well...EW IS a carry...
+-- LongCarry 3 -> Better played Carry on Short lane than long
+
+object.heroName = 'Hero_EmeraldWarden'
+
+----------------------------------
+--	Warden items
+----------------------------------
+--[[ list code:
+	"# Item" is "get # of these"
+	"Item #" is "get this level of the item" --]]
+behaviorLib.StartingItems = {"Item_HealthPotion", "Item_RunesOfTheBlight", "2 Item_ManaPotion", "Item_GuardianRing"} -- Items: Health Potion, Runes Of The Blight, 2 x Mana Potions, Guardian Ring
+behaviorLib.LaneItems = {"Item_Marchers", "Item_EnhancedMarchers"} 													 -- Items: Marchers -> Ghost Marchers
+behaviorLib.MidItems = {"Item_Protect", "Item_ArclightCrown", "Item_Critical1 4" } 									 -- Items: Null Stone, Arclight Crown, Riftshards Lvl 4
+behaviorLib.LateItems = {"Item_LifeSteal4", "Item_Evasion"} 														 -- Items: Symbol Of Rage, Wingbow
+
+--------------------------------
+-- Skills
+--------------------------------
+local bSkillsValid = false
+object.tSkills = {
+    1, 0, 1, 2, 	-- Levels 1-4 -> Max Wolves skill for dmg output with one point in Silence and one point in Overgrowth for escape
+	1, 3, 1, 0,     -- Levels 5-8 -> Then continue with maxing Silence
+	0, 0, 3, 2,     -- Levels 9-12 -> Then finish maxing Overgrowth
+	2, 2, 4, 		-- Levels 13-15 -> 
+	3, 4, 4, 4, 4, 4, 4, 4, 4, 4,	-- Levels 17-25 -> Attribute Points
+}
+function object:SkillBuild()
+
+	local unitSelf = self.core.unitSelf
+
+	if not bSkillsValid then
+		skills.abilSilence = unitSelf:GetAbility(0)   -- 1st Skill -> Silencing Shot
+		skills.abilWolves = unitSelf:GetAbility(1)  -- 2nd Skill -> Hunter's Command
+		skills.abilOvergrowth = unitSelf:GetAbility(2)  -- 3rd Skill -> Overgrowth
+		skills.abilGawain = unitSelf:GetAbility(3)  -- Ultimate -> Summon Gawain
```

Comment:
they're all passive so i don't have to use them... i should only add the threshold for them for increased aggressiveness?